modules/ossm-exposing-a-service-by-using-the-kubernetes-gateway-api-in-ambient-mode.adoc

```diff
@@ -0,0 +1,304 @@
+// Module included in the following assemblies:
+// * service-mesh-docs-main/gateways/ossm-getting-traffic-into-a-mesh.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ossm-exposing-a-service-by-using-the-kubernetes-gateway-api-in-ambient-mode_{context}"]
+= Exposing a service by using the {k8s} Gateway API in ambient mode
+
+[role="_abstract"]
+You can use the {k8s} Gateway API to create `Gateway` and `HTTPRoute` resources and deploy a gateway in ambient mode. The resources configure the gateway to expose a service in the mesh to traffic outside the mesh.
+
+.Prerequisites
+
+* You are logged in to the {ocp-product-title} web console as a user with the `cluster-admin` role.
+
+* You installed the {SMProductName} Operator.
+
+* You have deployed the `{istio}` resource.
+
+* You are using the {istio} ambient mode.
+
+* You use the {k8s}-native Gateway API resources.
+
+* You are either using the {istio} ambient mode or planning on migrating to the ambient mode.
+
+[NOTE]
+====
+If you are using the ambient mode (`istio.io/dataplane-mode=ambient`), you must use the {k8s} Gateway API approach. {istio} `Gateway` and `VirtualService` resources are not compatible with ambient mode.
+====
+
+.Procedure 
+
+. Create a namespace called `httpbin` by running the following command:
++
+[source,terminal]
+----
+$ oc create namespace httpbin 
+----
+
+. Apply the label for ambient mode by running the following command:
++
+[source,terminal]
+----
+$ oc label namespace httpbin istio.io/dataplane-mode=ambient
+----
+
+. Deploy a sample service named `httpbin` by running the following command:
++
+[source,terminal]
+----
+$ oc apply -n httpbin -f https://raw.githubusercontent.com/openshift-service-mesh/istio/refs/heads/master/samples/httpbin/httpbin.yaml
+----
+
+. Deploy a waypoint proxy by creating a YAML file named `httpbin-waypoint.yaml` that defines a {k8s}Gateway resource, similar to the following example:
++
+[source,yaml]
+----
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: httpbin-waypoint
+  namespace: httpbin
+  labels:
+    istio.io/waypoint-for: service
+spec:
+  gatewayClassName: istio-waypoint
+  listeners:
+  - name: mesh
+    port: 15008
+    protocol: HBONE
+
+----
+
+. Apply the YAML file by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f httpbin-waypoint.yaml
+----
+
+. Enable ingress waypoint routing on the `httpbin` service by running the following command:
++
+[source,terminal]
+----
+$ oc label service httpbin -n httpbin istio.io/ingress-use-waypoint=true
+----
++
+The label ensures that traffic from the ingress gateway routes through the waypoint proxy and the L7 policies configured on the waypoint proxy are applied to the ingress traffic, before reaching the `httpbin` service.
+
```

Comment:
The following step is missing, please add this.

Label the namespace so that all the services inside the namespace will be routed through the waypoint.

```
oc label ns httpbin istio.io/use-waypoint=httpbin-waypoint
```